source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRImageRenderer.mm

```diff
@@ -74,6 +73,24 @@ - (UIView *)render:(UIView<ACRIContentHoldingView> *)viewGroup
 
     if(img) {// if image is ready, proceed to add it
         view.image = img;
+        if(imgElem->GetImageSize() == ImageSize::Auto || imgElem->GetImageSize() == ImageSize::Stretch || imgElem->GetImageSize() == ImageSize::None){
+            CGFloat heightToWidthRatio = img.size.height / img.size.width;
+            [view addConstraints:@[[NSLayoutConstraint constraintWithItem:view
+                                                                    attribute:NSLayoutAttributeHeight
+                                                                    relatedBy:NSLayoutRelationEqual
+                                                                       toItem:view
+                                                                    attribute:NSLayoutAttributeWidth
+                                                                   multiplier:heightToWidthRatio
+                                                                     constant:0]]];
+            CGFloat widthToHeightRatio = img.size.width/ img.size.height;
+            [view addConstraints:@[[NSLayoutConstraint constraintWithItem:view
+                                                                    attribute:NSLayoutAttributeWidth
+                                                                    relatedBy:NSLayoutRelationEqual
+                                                                       toItem:view
+                                                                    attribute:NSLayoutAttributeHeight
+                                                                   multiplier:widthToHeightRatio
+                                                                     constant:0]]];
```

Comment:
Do you need both of these constraints? It looks like they do the same thing.

Reply:
Yes, it installs the constraint to each dimension.
if only one dimension has a constraint, and have dimension changes on the other dimension, it won't get updated.

---
In reply to: [175207669](https://github.com/Microsoft/AdaptiveCards/pull/1264#discussion_r175207669) [](ancestors = 175207669)